src/Engine/ProtoCore/Namespace/ElementRewriter.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System.Diagnostics;
+using ProtoCore.AST;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using ProtoCore.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ProtoCore.Namespace
+{
+
+    public class ElementRewriter
+    {
+        private readonly ElementResolver elementResolver;
+
+        private ElementRewriter(ElementResolver elementResolver)
+        {
+            this.elementResolver = elementResolver;
+        }
+
+        /// <summary>
+        /// Lookup namespace resolution map to substitute 
+        /// partial classnames with their fully qualified names in ASTs.
+        /// If partial class is not found in map, 
+        /// update ResolutionMap with fully resolved name from compiler.
+        /// </summary>
+        /// <param name="classTable"></param>
+        /// <param name="elementResolver"></param>
+        /// <param name="astNodes"> parent AST node </param>
+        public static void ReplaceClassNamesWithResolvedNames(ClassTable classTable,
+            ElementResolver elementResolver, ref IEnumerable<Node> astNodes)
+        {
+            var elementRewriter = new ElementRewriter(elementResolver);
+
+            for (int i = 0; i < astNodes.Count(); ++i)
+            {
+                var astNode = astNodes.ElementAt(i) as AssociativeNode;
+                if(astNode == null)
+                    continue;
+
+                elementRewriter.LookupResolvedNameAndRewriteAst(classTable, ref astNode);
+            }
+        }
+
+        private void LookupResolvedNameAndRewriteAst(ClassTable classTable, ref AssociativeNode astNode)
```

Comment:
Having this as an instance method is weird, can we make it `static` like all other methods?